packages/desktop-client/src/components/reports/reports/Calendar.tsx

```diff
@@ -0,0 +1,958 @@
+              <MobileBackButton onPress={() => navigate('/reports')} />
+            }
+          />
+        ) : (
+          <PageHeader
+            title={
+              widget ? (
+                <EditablePageHeaderTitle
+                  title={title}
+                  onSave={onSaveWidgetName}
+                />
+              ) : (
+                title
+              )
+            }
+          />
+        )
+      }
+      padding={0}
+    >
+      <Header
+        allMonths={allMonths}
+        start={start}
+        end={end}
+        mode={mode}
+        onChangeDates={onChangeDates}
+        filters={conditions}
+        onApply={onApplyFilter}
+        onUpdateFilter={onUpdateFilter}
+        onDeleteFilter={onDeleteFilter}
+        conditionsOp={conditionsOp}
+        onConditionsOpChange={onConditionsOpChange}
+        show1Month={true}
+      >
+        {widget && (
+          <Button variant="primary" onPress={onSaveWidget}>
+            <Trans>Save widget</Trans>
+          </Button>
+        )}
+      </Header>
+      <View ref={refContainer as Ref<HTMLDivElement>} style={{ flexGrow: 1 }}>
+        <View
+          style={{
+            backgroundColor: theme.pageBackground,
+            paddingTop: 0,
+            minHeight: '350px',
+            overflowY: 'auto',
+          }}
+        >
+          <View
+            style={{
+              flexDirection: isNarrowWidth ? 'column-reverse' : 'row',
+              justifyContent: 'flex-start',
+              flexGrow: 1,
+              gap: 16,
+              position: 'relative',
+              marginBottom: 16,
+            }}
+          >
+            {data && (
+              <View
+                ref={mergedRef}
+                style={{
+                  flexGrow: 1,
+                  flexDirection: 'row',
+                  gap: '20px',
+                  overflow: 'auto',
+                  height: '100%',
+                  justifyContent: flexAlignment,
+                  display: 'flex',
+                  ...styles.horizontalScrollbar,
+                }}
+              >
+                {data.calendarData.map((calendar, index) => (
+                  <CalendarWithHeader
+                    key={index}
+                    calendar={calendar}
+                    totalIncome={totalIncome}
+                    totalExpense={totalExpense}
+                    onApplyFilter={onApplyFilter}
+                    firstDayOfWeekIdx={firstDayOfWeekIdx}
+                  />
+                ))}
+              </View>
+            )}
+            <CalendarCardHeader
+              start={start}
+              end={end}
+              totalExpense={totalExpense}
+              totalIncome={totalIncome}
+              isNarrowWidth={isNarrowWidth}
+            />
+          </View>
+        </View>
+        <SelectedProviderWithItems
+          name="transactions"
+          items={[]}
+          fetchAllIds={async () => {
+            return Promise.resolve([]);
```

Comment:
Do we actually need Promise.resolve here or can we just return an empty array?

Reply:
changed